portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,183 @@
+    // Get the maximum amount of comments to display data from the server.
+    int maxCommentsObtained = getMaxComments(request);
+    if (previousMax == 1 && maxCommentsObtained != -1){
+      previousMax = maxCommentsObtained;
+      System.err.println("ntarn debug previous max change to current max: " + maxCommentsObtained);
+    }
+    else if (maxCommentsObtained == -1){
+      maxCommentsObtained = previousMax;
+      System.err.println("ntarn debug previous max: " + maxCommentsObtained);
+    }
+    else{
+      System.err.println("ntarn debug previous max reach else statement: " + maxCommentsObtained);
+    }
+
+    // Prepare a Query instance with the Comment kind of entity to load.
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Get the maximum amount of comments that can be displayed on a page.
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+      String imageUrl = (String) entity.getProperty("imageUrl");
+      Comment comment = new Comment(id, text, timestamp, imageUrl);
+      comments.add(comment);
+      if(comments.size() >= maxCommentsObtained){
+        break;
+      }
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the comment text input from the form.
+    String text = request.getParameter("text-input");
+    List<String> parameterNamesList =   Collections.list(request.getParameterNames());
+    long timestamp = System.currentTimeMillis();
+    String imageUrl = "";
+
+    // Get the URL of the image that the user uploaded to Blobstore.
+    if(request.getParameter("image") != null){
+      imageUrl = getUploadedFileUrl(request, "image");
+    }
+
+    // Create an Entity for the comment that can be entered into the DataStore.
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", text);
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("imageUrl", imageUrl);// == null ? "" : imageUrl); //separate this somehow
+
+    // Put newly created Entity.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/comments.html");
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
```

Comment:
This is the convention we follow here at Google.
Use the same concept everywhere for javadoc comments where you refer to `null`
```suggestion
  /** Returns a URL that points to the uploaded file, or {@code null} if the user didn't upload a file. */
```